src/docs/cmdstan-guide/json-format.tex

```diff
@@ -0,0 +1,111 @@
+\chapter{JSON Format}\label{json.chapter}
+
+\noindent 
+\CmdStan can use JSON format for input data for both
+model data and parameters. Model data is read in by the model
+constructor. Model parameters are used to initialize the sampler and
+optimizer.
+
+\section{Stan Data Types in JSON Notation}
+
+A Stan input file in  JSON notation consists of  JSON object which contains zero
+or more name-value pairs.  This structure corresponds to a Python data
+dictionary object.  The following is an example of JSON data for the
+simple Bernoulli example model:
+
+%
+\begin{quote}
+\begin{Verbatim}
+{ "N" : 10, "y" : [ 0, 1, 0, 0, 0, 0, 0, 0, 0, 1 ] }
+\end{Verbatim}
+\end{quote}
+%
+
+Matrix data and multi-dimensional arrays are indexed in row-major
+order.  For a Stan program which has data block
+\begin{quote}
+\begin{Verbatim}
+data {
+  int d1;
+  int d2;
+  int d3;
+  int ar[d1, d2, d3];
+}
+\end{Verbatim}
+\end{quote}
+\noindent
+the following JSON input file would be valid
+\begin{quote}
+\begin{Verbatim}
+{
+    "d1": 2,
+    "d2" : 3,
+    "d3" : 4,
+    "ar": [[[0, 1, 2, 3], [4, 5, 6, 7], [8, 9, 10, 11]], 
+            [[12, 13, 14, 15], [16, 17, 18, 19], [20, 21, 22, 23]]]
+}
+\end{Verbatim}
+\end{quote}
+
+All data inputs are encoded as name-value pairs.
+The following table provides more examples of JSON data.
+The left column contains a Stan data variable declaration
+and the right column contains valid JSON data inputs.
+%
+\begin{center}
+\begin{tabular}{r||c|c}
+{\it Stan variable} & {\it JSON data} \\ \hline \hline
+\code{int i;} & \code{"i" : 17} \\
+\\ 
+\code{real a;} & \code{"a" : 17} \\
+ & \code{"a" : 17.2} \\
+ & \code{"a" : "NaN"} \\
+ & \code{"a" : "+inf"} \\
+ & \code{"a" : "-inf"} \\
+\\
+\code{int a[5];} & \code{"a" : [1, 2, 3, 4, 5]} \\
+\\
+\code{real a[5];} & \code{"a" : [ 1, 2, 3.3, "NaN", 5 ]} \\
+\code{vector[5] a;} & \code{"a" : [ 1, 2, 3.3, "NaN", 5 ]} \\
+\code{row\_vector[5] a;} & \code{"a" : [ 1, 2, 3.3, "NaN", 5 ]} \\
+\code{real a[5];} & \code{"a" : [ 1, 2, 3.3, "NaN", 5 ]} \\
+\\
+\code{matrix[2,3] a;} & \code{"a" : [ [ 1, 2, 3 ], [ 4, 5, 6] ]} \\
+\end{tabular}
+\end{center}
+%
+
+
+\section{JSON Syntax Summary}
+
+JSON is a data interchange notation, defined by an 
```

Comment:
I think the standards reference should be moved to the top of the section and it should be stressed that we're following the standard so that standard output should be parsable.

Reply:
done